tests/phpunit/tests/html-api/wpHtmlProcessorHtml5lib.php

```diff
@@ -54,10 +54,16 @@ class Tests_HtmlApi_Html5lib extends WP_UnitTestCase {
 	 * @param string $expected_tree    Tree structure of parsed HTML.
 	 */
 	public function test_parse( ?string $fragment_context, string $html, string $expected_tree ) {
-		$processed_tree = self::build_tree_representation( $fragment_context, $html );
+		try {
+			$processed_tree = self::build_tree_representation( $fragment_context, $html );
+		} catch ( WP_HTML_Unsupported_Exception $e ) {
+			$this->markTestSkipped( "Unsupported markup: {$e->getMessage()}" );
+			return;
+		}
 
 		if ( null === $processed_tree ) {
 			$this->markTestSkipped( 'Test includes unsupported markup.' );
```

Comment:
This is probably inaccurate now, the unsupported markup messages should appear above.

I'd like to improve the other error cases because returning `null` can't express all of them. I've been thinking about adding more classes to handle them.